### ISSUE TITLE
UglifyJS overmangles `envCachesTemplates`

### ISSUE THREAD

**Issue by kaste** [Polymer/lit-html#223]
Okay, in

```
const envCachesTemplates =
    ((t) => t() === t())(() => ((s) => s) ``)
```

Out: 

```
const envCachesTemplates=(e=>e()==e())(()=>e=>e``);
```

which will evaluate to `false`. 

![image](https://user-images.githubusercontent.com/8558/33840739-125a927a-de96-11e7-9f89-51740b955164.png)

**Comment by Lodin** [comment]
It seems to be fixed in [TypeScript v2.6](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#cache-tagged-template-objects-in-modules). Shouldn't this fix be removed to reduce library size?

**Comment by justinfagnani** [comment]
We're just removing `envCachesTemplates` now that TypeScript fixed template literals in 2.6
